contrib/poly2tri/poly2tri/sweep/sweep_context.cc

```diff
@@ -96,11 +96,11 @@ void SweepContext::InitTriangulation()
 
 void SweepContext::InitEdges(std::vector<Point*> polyline)
 {
-  int num_points = polyline.size();
-  for (int i = 0; i < num_points; i++) {
-    int j = i < num_points - 1 ? i + 1 : 0;
-    edge_list.push_back(new Edge(*polyline[i], *polyline[j]));
+  std::size_t num_points = polyline.size();
+  for (std::size_t i = 0; i < num_points - 1; i++) {
+    edge_list.push_back(new Edge(*polyline[i], *polyline[i + 1]));
   }
+  edge_list.push_back(new Edge(*polyline.back(), *polyline.front()));
```

Comment:
I guess there's an implicit assumption that polyline is non-empty? Is this enforced/validated anywhere? Just asking, as the previous version handled that case gracefully (i.e., it won't crash), whereas this may not.